src/lib/y2network/interface_config_builder.rb

```diff
@@ -0,0 +1,160 @@
+    end
+
+    # Stores one option value tuple
+    # FIXME: obsolete by []=
+    def set(option: nil, value: nil)
+      @config[option] = value
+    end
+
+    # Returns currently stored option value
+    # FIXME: obsolete by []
+    def option(option)
+      @config.fetch(option, nil)
+    end
+
+    def []=(key, value)
+      @config[key] = value
+    end
+
+    def [](key)
+      @config[key]
+    end
+
+    # Provides stored configuration in sysconfig format
+    #
+    # @return [Hash<String, String>] where key is sysconfig option and value is the option's value
+    def device_sysconfig
+      # with naive implementation of filtering options by type
+      config = @config
+
+      # initalize options which has to be known and was not set by the user explicitly
+      init_mandatory_options
+
+      # filter out options which are not needed
+      config = config.delete_if { |k, _| k =~ /WIRELESS.*/ } if type != "wlan"
+      config = config.delete_if { |k, _| k =~ /BONDING.*/ } if type != "bond"
+      config = config.delete_if { |k, _| k =~ /BRIDGE.*/ } if type != "br"
+      config = config.delete_if { |k, _| k =~ /TUNNEL.*/ } if !["tun", "tap"].include?(type)
+      config = config.delete_if { |k, _| k == "VLAN_ID" || k == "ETHERDEVICE" } if type != "vlan"
+      config = config.delete_if { |k, _| k == "IPOIB_MODE" } if type != "ib"
+      config = config.delete_if { |k, _| k == "INTERFACE" } if type != "dummy"
+      config = config.delete_if { |k, _| k == "IFPLUGD_PRIORITY" } if config["STARTMODE"] != "ifplugd"
+
+      # all keys / values has to be strings, agent won't write it otherwise
+      config.map { |k, v| [k, v.to_s] }.to_h
+    end
+
+    # Updates itself according to the given sysconfig configuration
+    #
+    # @param devmap [Hash<String, String>] a key, value map where key is sysconfig option
+    #                                      and corresponding value is the option value
+    def load_sysconfig(devmap)
+      @config = !devmap.nil? ? @config.merge(devmap) : @config
+    end
+
+    def load_s390_config(devmap)
+      @s390_config = !devmap.nil? ? @s390_config.merge(devmap) : @s390_config
+    end
+
+  private
+
+    # returns a map with device options for newly created item
+    def init_mandatory_options
+      # FIXME: NetHwDetection is done in Lan.Read
+      Yast.import "NetHwDetection"
+
+      # #104494 - always write IPADDR+NETMASK, even empty
+      # #50955 omit computable fields
+      @config["BROADCAST"] = ""
+      @config["NETWORK"] = ""
+
+      if !@config["NETMASK"] || @config["NETMASK"].empty?
+        @config["NETMASK"] = Yast::NetHwDetection.result["NETMASK"] || "255.255.255.0"
+      end
+
+      @config["STARTMODE"] = new_device_startmode if !@config["STARTMODE"] || @config["STARTMODE"].empty?
+    end
+
+    # returns default startmode for a new device
+    #
+    # startmode is returned according product, Arch and current device type
+    def new_device_startmode
```

Comment:
I do not get why it is rewritten here from InterfaceDefaults.

Reply:
see above it needs some polishing. Most probably it used to be initialized to a default originally and later someone designed this product based proposal.